### PR TITLE
Block on dl only

### PR DESCRIPTION
**Pull request by wsorenson**
Allows tar to happen in parallels.

### CODE REVIEW

File: SingularityS3Base/src/main/java/com/hubspot/singularity/s3/base/ArtifactDownloadRequest.java [hunk 1]
```diff
@@ -4,21 +4,25 @@
...
   @JsonCreator
-  public ArtifactDownloadRequest(@JsonProperty("targetDirectory") String targetDirectory, @JsonProperty("s3Artifact") S3Artifact s3Artifact) {
+  public ArtifactDownloadRequest(@JsonProperty("targetDirectory") String targetDirectory, @JsonProperty("s3Artifact") S3Artifact s3Artifact,
+      @JsonProperty("timeoutMillis") Optional<Long> timeoutMillis)  {
     Preconditions.checkNotNull(targetDirectory);
     Preconditions.checkNotNull(s3Artifact);
 
     this.targetDirectory = targetDirectory;
     this.s3Artifact = s3Artifact;
+    this.timeoutMillis = timeoutMillis;
```
Comment (aaylward) [inline SingularityS3Base/src/main/java/com/hubspot/singularity/s3/base/ArtifactDownloadRequest.java]: checkNotNull?

Comment (aaylward) [inline SingularityS3Base/src/main/java/com/hubspot/singularity/s3/base/ArtifactDownloadRequest.java]: oh word 👍

**Comment by ssalinas** [comment]
@wsorenson or @tpetr is this PR gtg to stable?

**Comment by tpetr** [comment]
Should be good to merge to hs_stable

**Comment by ssalinas** [comment]
This has been working fine in hs_stable, going to merge

**Pull request merged**